src/main/kotlin/nmcp/NmcpAggregationExtension.kt

```diff
@@ -0,0 +1,80 @@
+package nmcp
+
+import gratatouille.GExtension
+import javax.inject.Inject
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpConsumerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.file.ArchiveOperations
+import org.gradle.api.tasks.bundling.Zip
+
+@GExtension(pluginId = "com.gradleup.nmcp.aggregation")
+abstract class NmcpAggregationExtension(private val project: Project) {
+    internal val spec = project.objects.newInstance(CentralPortalOptions::class.java)
+
+    @get:Inject
+    abstract val archiveOperations: ArchiveOperations
+
+    internal val consumerConfiguration = project.configurations.create(nmcpConsumerConfigurationName) {
+        it.isCanBeResolved = true
+        it.isCanBeConsumed = false
+
+        it.configureAttributes(project)
+    }
+
+    init {
+        val operations = archiveOperations
+        val layout = project.layout
+        val files= project.files(consumerConfiguration)
```

Comment:
`project.files(consumerConfiguration)` seems redundand as `consumerConfiguration.elements` is just fine below, isn't it?

Reply:
It uses `files.elements` (on line 35). CC doesn't like to serialize configurations. My understanding is that it needs to go through files.